storage/local/storage.go

```diff
@@ -336,67 +336,96 @@ func (s *memorySeriesStorage) NewPreloader() Preloader {
 	}
 }
 
-// FingerprintsForLabelMatchers implements Storage.
-func (s *memorySeriesStorage) FingerprintsForLabelMatchers(labelMatchers metric.LabelMatchers) clientmodel.Fingerprints {
+// FingerprintsForLabels implements Storage.
+func (s *memorySeriesStorage) FingerprintsForLabels(pairs ...metric.LabelPair) map[clientmodel.Fingerprint]struct{} {
 	var result map[clientmodel.Fingerprint]struct{}
-	for _, matcher := range labelMatchers {
+	for _, pair := range pairs {
 		intersection := map[clientmodel.Fingerprint]struct{}{}
-		switch matcher.Type {
-		case metric.Equal:
-			fps, err := s.persistence.fingerprintsForLabelPair(
-				metric.LabelPair{
-					Name:  matcher.Name,
-					Value: matcher.Value,
-				},
-			)
-			if err != nil {
-				log.Error("Error getting fingerprints for label pair: ", err)
-			}
-			if len(fps) == 0 {
-				return nil
-			}
-			for _, fp := range fps {
-				if _, ok := result[fp]; ok || result == nil {
-					intersection[fp] = struct{}{}
-				}
+		fps, err := s.persistence.fingerprintsForLabelPair(pair)
+		if err != nil {
+			log.Error("Error getting fingerprints for label pair: ", err)
+		}
+		if len(fps) == 0 {
+			return nil
+		}
+		for _, fp := range fps {
+			if _, ok := result[fp]; ok || result == nil {
+				intersection[fp] = struct{}{}
 			}
-		default:
-			values, err := s.persistence.labelValuesForLabelName(matcher.Name)
-			if err != nil {
-				log.Errorf("Error getting label values for label name %q: %v", matcher.Name, err)
+		}
+		if len(intersection) == 0 {
+			return nil
+		}
+		result = intersection
+	}
+	return result
+}
+
+// MetricsForLabelMatchers returns the metrics from storage that satisfy the given label matchers.
+func MetricsForLabelMatchers(storage Storage, matchers ...*metric.LabelMatcher) map[clientmodel.Fingerprint]clientmodel.COWMetric {
+	var (
+		equals  []metric.LabelPair
```

Comment:
We have different perceptions then. We should still agree on one style vs. the other generally. Doesn't have to be now, because there's more important things to worry about...